pkg/cmdimpl/add.go

```diff
@@ -349,134 +270,276 @@ func Add(args []string, allParams AddParamSet, deps *AddDependencies) error {
-			return wrapError(err, fmt.Sprintf("could not clone repository: %s", wegoRepoURL))
-		}
+	goatRepoSource, err := generateSource(goatSourceName, params.AppConfigUrl, SourceTypeGit)
+	if err != nil {
+		return err
+	}
+	goatTargetKustomize, err := generateTargetKustomize(goatSourceName, ".")
+	if err != nil {
+		return err
+	}
+	goatAppKustomize, err := generateAppKustomize(goatSourceName, ".")
+	if err != nil {
+		return err
+	}
+	if err := applyToCluster(goatRepoSource, goatTargetKustomize, goatAppKustomize); err != nil {
+		return err
+	}
+	// Source covers entire user repo
+	userSourceName := getUserRepoName()
+	userRepoSource, err := generateSource(userSourceName, params.Url, SourceType(params.SourceType))
+	if err != nil {
+		return err
+	}
+	userTargetKustomize, err := generateTargetKustomize(userSourceName, ".")
+	if err != nil {
+		return err
+	}
+	userAppKustomize, err := generateAppKustomize(userSourceName, ".")
+	if err != nil {
+		return err
 	}
+	appYaml, err := generateAppYaml()
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create app.yaml for '%s'", params.Name))
+	}
+	applicationGoat, err := generateApplicationGoat(userSourceName)
+	if err != nil {
+		return wrapError(err, fmt.Sprintf("could not create GitOps automation for '%s'", params.Name))
+	}
+	if err := writeAppYaml(gitClient, appYaml, "."); err != nil {
+		return err
+	}
+	if err := writeGoats(gitClient, "", userRepoSource, userTargetKustomize, userAppKustomize, applicationGoat); err != nil {
+		return err
+	}
+	return commitAndPush(ctx, gitClient)
+}
 
-	fmt.Fprintf(shims.Stdout(), "Applying wego platform resources...\n")
-	if !params.DryRun {
-		// Install Source and Kustomize controllers, and CRD for application (may already be present)
-		wegoSource, err := generateWegoSourceManifest()
-		if err != nil {
-			return wrapError(err, "could not generate wego source manifest")
-		}
+func generateSource(repoName, repoUrl string, sourceType SourceType) ([]byte, error) {
+	secretName := repoName
+
+	cmd := fmt.Sprintf(`create secret git "%s" \
+            --url="%s" \
+            --private-key-file="%s" \
+            --namespace="%s"`,
+		secretName,
+		repoUrl,
+		params.PrivateKey,
+		params.Namespace)
+	if params.DryRun {
+		fmt.Printf(cmd + "\n")
+	} else {
+		_, err := fluxops.CallFlux(cmd)
 
-		wegoKust, err := generateWegoKustomizeManifest()
 		if err != nil {
-			return wrapError(err, "could not generate wego kustomize manifest")
+			return nil, wrapError(err, "could not create git secret")
 		}
+	}
 
-		kubectlApply := fmt.Sprintf("kubectl apply --namespace=%s -f -", params.Namespace)
+	cmd = fmt.Sprintf(`create source git "%s" \
+            --url="%s" \
+            --branch="%s" \
+            --secret-ref="%s" \
+            --interval=30s \
+            --export \
+            --namespace="%s"`,
+		repoName,
+		repoUrl,
+		params.Branch,
+		secretName,
+		params.Namespace)
+	sourceManifest, err := fluxops.CallFlux(cmd)
```

Comment:
Missing the params.DryRun check - so in dry-run mode we will try and create the source but the secret won't be there.

Reply:
We don't actually create the source; we just create the manifest and return it.